ipod-shuffle-4g.py

```diff
@@ -562,6 +589,9 @@ def construct(self, tracks):
         for i in self.listtracks:
             path = self.ipod_to_path(i)
             position = -1
+            if self["listtype"] == 1 and "/iPod_Control/Podcasts/" in path:
+                print ('not including podcast in master playlist: {}'.format(path))
```

Comment:
Do we really need to print that to the user? Also shouldnt the listtype be 3 instead of 1?

Reply:
I can remove that printout. The intention was to match the print level of other messages.
Also, this section is excluding podcasts from the All Songs playlist. That's a bit more clear with the enum, but I added a comment.

Reply:
I have not run the program for ages. If you say the print fits, we can keep it. But please use a capital `Not` at the beginning :-D

I currently try to understand the code again. Isnt there a track object inside that playlist of which we can check if it is a podcast. But at another point we also accept id3 tags with Genre set to Podcast, but this code here does not support this.

Reply:
It's not a track object, just a path to the track. In order to detect by id3 at this point, the code would have to reference the track header and look up each track object, or reload the metadata with mutagen. I think that would be better handled by a bigger re-architecture that loads all data, building the database objects and then writes everything at once, rather than building up some data as the database is built.